testing/fpgainfo/test_fpgainfo_c.cpp

```diff
@@ -483,6 +333,63 @@ TEST_P(fpgainfo_c_p, errors_command0) {
     fpgaDestroyProperties(&filter);
 }
 
+/**
+ * @test       parse_error_args_errors_clear
+ * @brief      Test: parse_error_args
+ * @details    When passed the clear errors option, the function <br>
+ *             prints and clears errors.<br>
+ */
+TEST_P(fpgainfo_c_p, parse_error_args_errors_clear) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+
+    fpga_properties filter = NULL;
+    fpga_token *tokens = NULL;
+    uint32_t matches = 0, num_tokens = 0;;
+
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+    ASSERT_EQ(fpgaEnumerate(&filter, 1, NULL, 0, &matches), FPGA_OK);
+    ASSERT_GT(matches, 0);
+    tokens = (fpga_token *)malloc(matches * sizeof(fpga_token));
+
+    num_tokens = matches;
+    ASSERT_EQ(fpgaEnumerate(&filter, 1, tokens, num_tokens, &matches), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "errors");
+    strcpy(two, "all");
+    strcpy(three, "-c");
+    EXPECT_EQ(parse_error_args(4, argv), 0);
+
+    EXPECT_EQ(errors_command(tokens, num_tokens, 4, argv), FPGA_OK);
```

Comment:
The parse_errors_args() must be called prior to errors_command() as it sets errors_config.clear static variable. Otherwise the call to errors_command() will not take the clear errors path when called. In the regular flow of the tool, this is how it would go - although it is kinda weird.

Reply:
👍